crates/frontend/src/components/context_card/mod.rs

```diff
@@ -0,0 +1,80 @@
+use leptos::*;
+use serde_json::{Map, Value};
+
+use crate::{
+    components::{
+        condition_pills::{condition_pills::ConditionPills, types::Condition},
+        table::{table::Table, types::Column},
+    },
+    types::Context,
+};
+
+#[component]
+pub fn context_card(
+    context: Context,
+    overrides: Vec<(String, Value)>,
+    handle_edit: Callback<(Context, Vec<(String, Value)>), ()>,
+    handle_clone: Callback<(Context, Vec<(String, Value)>), ()>,
+    handle_delete: Callback<String, ()>,
+) -> impl IntoView {
+    let conditions: Vec<Condition> = (&context).try_into().unwrap_or(vec![]);
+    let override_table_rows = overrides
+        .clone()
+        .into_iter()
+        .map(|(k, v)| {
+            let k = Value::String(k.trim_matches('"').to_string());
+            let v = Value::String(format!("{}", v).trim_matches('"').to_string());
+            Map::from_iter(vec![(String::from("KEY"), k), (String::from("VALUE"), v)])
+        })
+        .collect::<Vec<Map<String, Value>>>();
+
+    let context = StoredValue::new(context);
+    let overrides = StoredValue::new(overrides);
+
+    let table_columns = vec![
+        Column::default("KEY".to_string()),
+        Column::default("VALUE".to_string()),
+    ];
+
+    view! {
+        <div class="rounded-lg shadow bg-base-100 p-6 shadow">
+            <div class="flex justify-between">
+                <div class="flex items-center space-x-4">
+                    <h3 class="card-title text-base timeline-box text-gray-800 bg-base-100 shadow-md font-mono">
+                        "Condition"
+                    </h3>
+                    <i class="ri-arrow-right-fill ri-xl text-blue-500"></i>
+                    <ConditionPills conditions=conditions/>
+                </div>
+                <div class="flex space-x-4">
+                    <i
```

Comment:
{["pencil-line", "file-copy-line", "delete-bin-5-line"].map(|icon| {
            <i
                class=format!("ri-{} ri-xl text-blue-500 cursor-pointer", icon)
                on:click=move |_| {
                    match icon {
                        "pencil-line" => handle_edit.call((context.get_value(), overrides.get_value())),
                        "file-copy-line" => handle_clone.call((context.get_value(), overrides.get_value())),
                        "delete-bin-5-line" => handle_delete.call(context.get_value().id),
                        _ => {}
                    }
                }
            >
            </i>
        })}
        This will decrease the DRY, the html <i> tag was repeating here.

Reply:
This way of writing code really un-intuitive, DRY should be a goal, but here its not needed, its unnecessarily increasing code complexity.

```
match icon { "pencil-line" => handle_edit.call((context.get_value(), overrides.get_value())), "file-copy-line" => handle_clone.call((context.get_value(), overrides.get_value())), "delete-bin-5-line" => handle_delete.call(context.get_value().id), _ => {} } 
```
Doing a match to put the handlers, makes unnecessary complex code.

Reply:
Sure , It affects the readability so we can keep as it is .